File: Practicals/07_11/calc.java
/*
 *******************************************************************************
 *  W.A.P. to make a menu driven Calculator
 *******************************************************************************
 */

import java.util.Scanner;


class Calc{

    private float answer;

    float getAnswer() {
        return answer;
    }
    void setAnswer(float x){
        answer = x;
    }

    void addition(float x, float y) {
        float ans;
        ans = x + y;
        setAnswer(ans);
    }
    void subtraction(float x, float y) {
        float ans;
        ans = x - y;
        setAnswer(ans);
    }
    void multiplication(float x, float y) {
        float ans;
        ans = x * y;
        setAnswer(ans);
    }
    void division(float x, float y) {
        float ans;
        ans = x / y;
        setAnswer(ans);
    }
}

class Main {
    public static void main(String[] args) {
        boolean isDone = true;
        Scanner number = new Scanner(System.in);

        System.out.print("Enter First number : ");
        float num = number.nextFloat();
        System.out.print("Enter Second number : ");
        float num2 = number.nextFloat();

        System.out.println("Numbers are : " + num + " and " + num2);

        Calc obj = new Calc();

        System.out.println("Enter the choice of Operation : ");
        System.out.println("\t1. Addition \n\t2.Subtraction \n\t3.Multiplication \n\t4.Division");
        int choice = number.nextInt();

        switch(choice) {
            case 1:
                obj.addition(num, num2);
                break;
            case 2:
                obj.subtraction(num, num2);
                break;
            case 3:
                obj.multiplication(num, num2);
                break;
            case 4:
                obj.division(num, num2);
                break;
            default:
                System.out.println("Invalid Choice.");
                isDone = false;
                break;
        }

        if(isDone)
            System.out.println("Answer is : " + obj.getAnswer());
    }
}

File: Practicals/07_11/Python/Calc.py

class Calc():
    def __init__(self):
        __answer = 0

    def setAnswer(self, x):
        self.__answer = x
    def getAnswer(self):
        return self.__answer

    def add(self, num1, num2):
        self.__answer = num1 + num2
    def sub(self,num1, num2):
        self.__answer = num1 - num2
    def mul(self, num1, num2):
        self.__answer = num1 * num2
    def div(self, num1, num2):
        self.__answer = num1 / num2

def main():
    num1 = int(input("Enter first number  : "))
    num2 = int(input("Enter second number : "))


    print("\t1. Addition \n\t2.Subtraction \n\t3.Multiplication \n\t4.Division")
    ch = input("Enter your choice : ")

    obj = Calc()

    if ch == '1':
        obj.add(num1, num2)
    elif ch == '2':
        obj.sub(num1, num2)
    elif ch == '3':
        obj.mul(num1, num2)
    elif ch == '4':
        obj.div(num1, num2)
    else:
        print("Invalid Choice.")

    print("answer is : ", obj.getAnswer())

if __name__ == "__main__":
    main()
